scraper/company_scrapper.go

```diff
@@ -0,0 +1,237 @@
+package scraper
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/google/go-querystring/query"
+	"github.com/spf13/viper"
+
+	"github.com/Businge931/company-email-scraper/configs"
+)
+
+// SerpAPI response struct visit: https://serper.dev/playground
+type SerpAPIResponse struct {
+	Organic []struct {
+		Link string `json:"link"`
+	} `json:"organic"`
+}
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+var (
+	// static error variables for GetSearchResults
```

Comment:
move all these errors into a module called `models` and within that module place them in a file `errors.go`